resources/tickets.py

```diff
@@ -1,32 +1,87 @@
-from flask_restful import Resource
-from models.tickets import TicketsModel
+from flask_restful import Resource, reqparse
+import json
+from models.tickets import TicketModel
+from models.notes import NotesModel
+from resources.admin_required import admin_required
+
 
 class Ticket(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument('sender')
+    parser.add_argument('tenant')
+    parser.add_argument('status')
+    parser.add_argument('urgency')
+    parser.add_argument('issue')
+    parser.add_argument('note')
+
+    @admin_required
     def get(self, id):
         ticket= TicketModel.find_by_id(id)
         if ticket:
             return ticket.json()
         return {'message': 'Ticket Not Found'}, 404
-    
-    def post(self,id):
-        if TicketModel.find_by_id(id):
-            return('message': 'Ticket already exists'), 400
-        
-        ticket = TicketModel(id)
-        try:
-            ticket.save_to_db()
-        except:
-            return {'Message': 'An Error Has Occured'}, 500
-
-        return ticket.json(), 201
 
+    @admin_required
     def delete(self, id):
         ticket= TicketModel.find_by_id(id)
         if ticket:
             ticket.delete_from_db()
 
         return{'Message': 'Ticket Removed from Database'}
 
+    @admin_required
+    def put(self, id):
+        data = Ticket.parser.parse_args()
+        ticket = TicketModel.find_by_id(id)
+
```

Comment:
Ideally, the ticket's existence should be verified before continuing to modify and save the updates. This can provide better error messages from the server.